assets/javascripts/discourse/components/chat-message.js

```diff
@@ -678,21 +701,19 @@ export default Component.extend({
 
   @action
   flag() {
-    bootbox.confirm(
-      I18n.t("chat.confirm_flag", {
-        username: this.message.user?.username,
-      }),
-      (confirmed) => {
-        if (confirmed) {
-          ajax("/chat/flag", {
-            method: "PUT",
-            data: {
-              chat_message_id: this.message.id,
-            },
-          }).catch(popupAjaxError);
-        }
-      }
-    );
+    const targetFlagSupported =
```

Comment:
How come this is required? Won't the core PR https://github.com/discourse/discourse/pull/18251 add this and get merged before this one?

Reply:
Oh sorry I saw the above. Hmm I wonder if there is another way we can do this rather than calling `requirejs.entries` directly. Maybe it's not a problem, if it's not don't worry about this comment. I can confirm it worked fine locally since I didn't have the core PR checked out when I tested this at first.

Reply:
Yeah, I tested it with and without the core refactor, and it works. I don't think there's a fancier way of doing it other than calling `require(...)` and catching the exception.

Can't you think of any reasons not to use `requirejs.entries`? I know it's hackish, but guaranteeing backward compatibility is tricky, and based on my experience so far, it's hard to come up with a generic solution for this problem. Also, this is much more common on the Ruby side.

Reply:
Yeah it's no problem it just stood out to me and it seemed hackish, but if we do it commonly and there's no better way then it's fine by me :+1: